### PR TITLE
Created a function setupMenu

### PR DESCRIPTION
**Pull request by momotav**

### CODE REVIEW

File: _client/src/Views/MainWindow.hpp [hunk 1]
```diff
@@ -20,13 +23,23 @@ class MainWindow : public QMainWindow
...
     QStackedWidget *m_stackView;
 
     // status bar
     QStatusBar *m_statusBar;
+
+    // menu bar
+    QMenuBar *m_mainMenuBar;
+
+    //elements of MenuBar
+    QMenu *m_fileMenu,*m_purchasingMenu,*m_salesMenu,*m_inventoryMenu,*m_analyticsMenu,*m_usersMenu,*m_managementMenu;
```
Comment (Sage-Cat) [inline _client/src/Views/MainWindow.hpp]: Remove this

File: _client/src/Views/MainWindow.cpp [hunk 2]
```diff
@@ -25,6 +28,51 @@ void MainWindow::setupUI()
     this->showMaximized();
 }
 
+void MainWindow::setupMenu() 
+{
+    // create MenuBar
+    m_mainMenuBar =  menuBar();
+    // File Menu
+    m_fileMenu = createModuleMenu(tr("File"), {ActionTypes::SETTINGS, ActionTypes::EXIT}); 
+    m_mainMenuBar->addMenu(m_fileMenu);
```
Comment (Sage-Cat) [inline _client/src/Views/MainWindow.cpp]: just `m_mainMenuBar->addMenu(createModuleMenu(tr("File"), {ActionTypes::SETTINGS, ActionTypes::EXIT}))`;

Remove instances of m_fileMenu and others

File: _client/src/Views/MainWindow.cpp [hunk 3]
```diff
@@ -25,6 +28,51 @@ void MainWindow::setupUI()
...
+
+    // Users Module Menu
+    m_usersMenu = createModuleMenu(tr("Users"), {ActionTypes::USER_ROLES, ActionTypes::USER_LOGS});
+    m_mainMenuBar->addMenu(m_usersMenu);
+
+    // Management Module Menu
+    m_managementMenu = createModuleMenu(tr("Management"), {ActionTypes::EMPLOYEES, ActionTypes::CUSTOMERS, ActionTypes::SUPPLIERS});
+    m_mainMenuBar->addMenu(m_managementMenu);
+}
+//Custom function that create module menu using map
```
Comment (Sage-Cat) [inline _client/src/Views/MainWindow.cpp:58]: new line

File: _client/src/Views/MainWindow.cpp [hunk 4]
```diff
@@ -25,6 +28,51 @@ void MainWindow::setupUI()
     this->showMaximized();
 }
 
+void MainWindow::setupMenu() 
+{
+    // create MenuBar
+    m_mainMenuBar =  menuBar();
+    // File Menu
```
Comment (Sage-Cat) [inline _client/src/Views/MainWindow.cpp]: new line

**Review by Sage-Cat — CHANGES_REQUESTED** [review]

File: _client/src/Views/MainWindow.hpp (hunk 1 above)
Comment (momotav) [reply _client/src/Views/MainWindow.hpp]: done

File: _client/src/Views/MainWindow.cpp (hunk 2 above)
Comment (momotav) [reply _client/src/Views/MainWindow.cpp]: done

File: _client/src/Views/MainWindow.cpp (hunk 3 above)
Comment (momotav) [reply _client/src/Views/MainWindow.cpp:58]: done

File: _client/src/Views/MainWindow.cpp (hunk 4 above)
Comment (momotav) [reply _client/src/Views/MainWindow.cpp]: done

File: _client/src/Views/MainWindow.cpp [hunk 5]
```diff
@@ -25,6 +28,47 @@ void MainWindow::setupUI()
...
+    m_mainMenuBar->addMenu(createModuleMenu(tr("Users"), {ActionTypes::USER_ROLES, ActionTypes::USER_LOGS}));
+
+    // Management Module Menu
+    m_mainMenuBar->addMenu(createModuleMenu(tr("Management"), {ActionTypes::EMPLOYEES, ActionTypes::CUSTOMERS, ActionTypes::SUPPLIERS}));
+}
+
+//Custom function that create module menu using map
+QMenu* MainWindow::createModuleMenu(const QString& menuTitle, const std::vector<ActionTypes>& actions)
+{
+    QMenu* menu = new QMenu(menuTitle);
```
Comment (Sage-Cat) [inline _client/src/Views/MainWindow.cpp:61]: Who manages lifetime of this object?
Make mainwindow it's parent

File: _client/src/Views/MainWindow.cpp [hunk 6]
```diff
@@ -25,6 +28,47 @@ void MainWindow::setupUI()
...
+
+    for (ActionTypes actionType : actions) {
+        QAction* action = new QAction(ACTION_NAMES.at(actionType).c_str(), this);
```
Comment (Sage-Cat) [inline _client/src/Views/MainWindow.cpp:64]: Let menu, that you create on line 61 manage lifetime of this actions. Isn't it logical?
Make menu a parent of it's actions

**Review by Sage-Cat — CHANGES_REQUESTED** [review]

File: _client/src/Views/MainWindow.cpp (hunk 5 above)
Comment (momotav) [reply _client/src/Views/MainWindow.cpp:61]: done

File: _client/src/Views/MainWindow.cpp (hunk 6 above)
Comment (momotav) [reply _client/src/Views/MainWindow.cpp:64]: done

**Pull request closed without merging**